packages/core/src/components/NetworkSwitch/NetworkSwitch.tsx

```diff
@@ -0,0 +1,59 @@
+import React, { useState, useEffect } from 'react';
+import { CHAIN_ID_TO_NETWORK } from '@web3-ui/hooks';
+import { Select, SelectProps } from '@chakra-ui/react';
+
+const useChainId = () => {
+  const [chainId, setChainId] = useState<null | string>(null);
+  useEffect(() => {
+    window.ethereum.on('chainChanged', setChainId);
+    window.ethereum.request({ method: 'eth_chainId' }).then(setChainId);
+  }, []);
+  return chainId;
+};
+
+const convertToHexStr = (num: number) => num.toString(16);
+
+export interface NetworkSwitchProps {
+  /**
+   * The chain IDs (in base 10) that the component should display as options
+   */
+  networks: number[];
+}
+
+/**
+ * A component to switch between a number of specified networks.
+ */
+export const NetworkSwitch: React.FC<NetworkSwitchProps & SelectProps> = ({
+  networks,
+  ...props
+}) => {
+  const chainId = useChainId();
+  const handleChange = (e: React.ChangeEvent<HTMLSelectElement>): void => {
+    const networkNum = Number(e.target.value);
+    switchNetwork(networkNum);
+  };
+  const switchNetwork = async (networkId: number) => {
+    const networkHex = `0x${convertToHexStr(networkId)}`;
+    if (window.ethereum) {
+      try {
+        await window.ethereum.request({
+          method: 'wallet_switchEthereumChain',
+          params: [{ chainId: networkHex }],
+        });
+      } catch (error) {
+        console.log(error);
+      }
+    } else {
+      alert(
+        'Switching networks automatically is only supported in MetaMask. Please consider installing it: https://metamask.io/download.html'
+      );
+    }
+  };
```

Comment:
This `switchNetwork()` logic is also used in `useWallet()`. Would it make more sense to consolidate the two into a single function in a common util file?

Reply:
I think it would, yes